clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2728,6 +2728,23 @@ static void emitKernelNameType(QualType T, ASTContext &Ctx, raw_ostream &OS,
   emitWithoutAnonNamespaces(OS, T.getCanonicalType().getAsString(TypePolicy));
 }
 
+int SYCLIntegrationHeader::getCppVersion() {
+  LangOptions LangOpts = S.getASTContext().getLangOpts();
+  if (LangOpts.CPlusPlus) {
+    if (LangOpts.CPlusPlus20)
+      return 201707L;
+    else if (LangOpts.CPlusPlus17)
+      return 201703L;
+    else if (LangOpts.CPlusPlus14)
+      return 201402L;
+    else if (LangOpts.CPlusPlus11)
+      return 201103L;
+    else
+      return 199711L;
+  }
+  return -1;
```

Comment:
This seems odd and misguided.  Having this function allow !CPlusPlus at all seems like a mistake.

Reply:
Updated